Update EIP-5453 (#6142)
ethereum/EIPs: EIPS/eip-5453.md

---
eip: 5453

> title: Endorse, Permit for Extendable Methods

Reviewer: ```suggestion
title: Endorsement Permits for Extendable Methods
```
How about this title?

## Abstract

> This EIP establish A general protocol for arbitrary number of approvers giving ECDSA permit in the same TX for any [EIP-5750](./eip-5750.md) conforming extendable methods. Unlike a few prior art ([EIP-2612](./eip-2612.md) for [EIP-20](./eip-20.md), `EIP-4494`(draft) for [EIP-721](./eip-721.md) that usually only permit for a single behavior (`transfer` for EIP-20 and `safeTransferFrom` for EIP-721) and a single approver in two transactions (first a `permit(...)` TX, then a `transfer`-like TX), this EIP provides a way to permit arbitrary behaviors and aggregating multiple approvals from arbitrary number of approvers in the same transaction, allowing for Multi-Sig or Threshold Signing behavior.

Reviewer: ```suggestion
This EIP establishes a general protocol for an arbitrary number of approvers to give ECDSA permits in the same transaction as any [EIP-5750](./eip-5750.md) conforming extendable methods. Unlike prior standards ([EIP-2612](./eip-2612.md) for [EIP-20](./eip-20.md) and [EIP-4494](./eip-4494.md) for [EIP-721](./eip-721.md) that usually only permit a single behavior (`transfer` for EIP-20 and `safeTransferFrom` for EIP-721) and a single approver using two transactions (first a `permit(...)` transaction, then a `transfer`-like transaction), this EIP provides a way to permit arbitrary behaviors and aggregate multiple approvals from an arbitrary number of approvers in the same transaction, allowing for multisig and threshold signing behavior.
```
Fixes a few typos

## Rationale

1. We chose to have both `ERC5453_TYPE_A`(single-endorsement) and `ERC5453_TYPE_B`(multiple-endorsements, same nonce for entire contract) so we

> could balance a wider range of use cases. E.g. the same use cases of EIP-2612 and `EIP-4494`(draft) can be supported by `ERC5453_TYPE_A`. And threshold approvals can be done via `ERC5453_TYPE_B`. More complicated approval types can also be extended by defining new `ERC5453_TYPE_?`

Reviewer: ```suggestion
could balance a wider range of use cases. E.g. the same use cases of EIP-2612 and EIP-4494 can be supported by `ERC5453_TYPE_A`. And threshold approvals can be done via `ERC5453_TYPE_B`. More complicated approval types can also be extended by defining new `ERC5453_TYPE_?`
```

> 1. We chose to include both `validSince` and `validBy` to allow maximum flexibility in expiration. This can be also be supported by EVM natively at  if adopted `EIP-5081`(draft) but it is not gaining consensus anytime soon, we choose to add these two numbers in this EIP to allow

Reviewer: ```suggestion
1. We chose to include both `validSince` and `validBy` to allow maximum flexibility in expiration. This can be also be supported by EVM natively at if adopted by [EIP-5081](./eip-5081.md). These two parameters allow for
```

### Reference Implementation of `EndorsableERC721`

> Here is a reference implementation of `EndorsableERC721` that achieves similar behavior of `EIP-4494`(draft).

Reviewer: ```suggestion
Here is a reference implementation of `EndorsableERC721` that achieves similar behavior as EIP-4494.
```
